prov/efa/src/efa_av.c

```diff
@@ -840,8 +840,8 @@ static int efa_av_close(struct fid *fid)
 	}
 
 	if (av->ep_type == FI_EP_RDM) {
-		if (rxr_env.enable_shm_transfer && av->shm_rdm_av &&
-		    &av->shm_rdm_av->fid) {
+		if (av->shm_rdm_av && &av->shm_rdm_av->fid) {
```

Comment:
Sorry, missed this earlier, isn't `&av->shm_rdm_av->fid` always going to be true based on the struct def? I think that can be dropped/wonder why this was there in the first place.

Reply:
You are totally right. Not sure of the origins of it but I've cleaned it up in a new push. Good catch!